tests/prop.rs

```diff
@@ -40,6 +56,17 @@ impl Language for Prop {
     fn cost(_node: &Expr<Prop, u64>) -> u64 {
         unimplemented!()
     }
+    fn eval(e: &RecExpr<Prop>) -> Bool {
+        match e.as_ref() {
+            Expr::Variable(_) => Bool::False, // TODO handle exception
+            Expr::Constant(c) => c.clone(),   // TODO should this be clone?
+            Expr::Operator(op, ns) => match op {
+                Op::And => ns.iter().map(Self::eval).fold(Bool::True, conjoin),
+                Op::Or => ns.iter().map(Self::eval).fold(Bool::False, disjoin),
+                _ => Bool::False, // TODO handle exception
```

Comment:
It might be easier to assert that the operator has the right number of children, and then just unpack it and implement it that way. Ex:
```rust
Op::Not => {
  assert_eq!(ns.len(), 1);
  if ns[0] == Bool::True { Bool::False } else { Bool::True }
}